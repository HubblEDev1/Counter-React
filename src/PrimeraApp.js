import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({saludo, subtitulo}) => {//destrecturing props

    return (
        <>
            <h1>{saludo}</h1>
            <p>{subtitulo}</p>
        </>
    );
}

PrimeraApp.prototype = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Mi primera app',
}

export default PrimeraApp;