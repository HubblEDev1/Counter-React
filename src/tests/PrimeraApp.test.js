//import React from 'react';
import '../setupTests';
import '@testing-library/jest-dom';

import {shallow} from 'enzyme';
import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";

describe('Puebas en <PrimeraApp/>', () => {
    /*test('Debe de mostrar un mensaje', ()=>{
        const greeting = 'Buenas tardes';
        //const wrapper = render( <PrimeraApp saludo={greeting}/> );
        //wrapper.getByText();
        const {getByText} = render( <PrimeraApp saludo={greeting}/> );
        
        expect(getByText(greeting)).toBeInTheDocument();
    })*/
    test('Debe de mostrar un mensaje', ()=>{
        const greeting = 'Buenas tardes';
        const wrapper = shallow(<PrimeraApp saludo={greeting}/>);//Its similar to render and allow us to render clicks

        expect(wrapper).toMatchSnapshot();//it allow us to modify the component


    });

    test('Debe de mostrar un mensaje', ()=>{
        const greeting = 'Buenas tardes';
        const subtitulo = 'SOy un subtitulo';

        const wrapper = shallow(
            <PrimeraApp 
                saludo={greeting}
                subtitulo={subtitulo}
            />
        );

        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe(subtitulo);//it allow us to modify the component
    });
});