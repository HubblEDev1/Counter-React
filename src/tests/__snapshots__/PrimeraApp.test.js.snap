// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en <PrimeraApp/> Debe de mostrar un mensaje 1`] = `
<Fragment>
  <h1>
    Buenas tardes
  </h1>
  <p>
    Mi primera app
  </p>
</Fragment>
`;
